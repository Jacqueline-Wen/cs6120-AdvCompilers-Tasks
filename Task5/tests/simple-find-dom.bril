@main {
    i: int = const 0;
    jmp .loop;

.loop:
    limit: int = const 10;
    cond: bool = lt i limit;
    br cond .body .exit;

.body:
    threshold: int = const 5;
    is_small: bool = lt i threshold;
    br is_small .then .endif;

.then:
    small_add: int = const 1;
    i: int = add i small_add;
    jmp .endif;

.endif:
    increment: int = const 1;
    i: int = add i increment;
    jmp .loop;

.exit:
    print i;
}