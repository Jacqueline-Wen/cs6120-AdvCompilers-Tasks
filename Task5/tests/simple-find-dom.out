Block 1 is dominated by: 1 
Block 2 is dominated by: 1 2 
Block 3 is dominated by: 1 2 3 
Block 4 is dominated by: 1 2 3 4 
Block 5 is dominated by: 1 2 3 5 
Block 6 is dominated by: 1 2 6 
